design/specs/column-extraction.md

```diff
@@ -8,87 +8,58 @@ editor: markdown
-Initially, just table splitting will be implemented, so that a user can select columns from an existing table and create a new one.
+## Context
 
-# Splitting a Table by Extracting Columns
-To create a new table from an existing one, select the columns to include in the new table. These columns must not be primary keys for the parent table (although they can be foreign keys to another table). Once selected, the inspector panel should display a list of possible 'Selection Actions,' with the 'New Table from Columns' option allowing the user to extract the columns into a new table.
+In data modeling, it's usual practice to split data up into related tables. In the context of Mathesar, one way of doing this is by extracting columns from a table into a new table. This process is known as column extraction.
 
-## Selecting Columns
-Columns must be given a new interaction in order to be selected. A column header can be selected by clicking on it, as in spreadsheet applications. Multi-selection is also possible by holding down the shift key and clicking on multiple column headers. Clearing the selection is as simple as clicking anywhere outside of the selection target or on a selected column once more.
+This spec describes the steps a user would take to create a new table from a subset of columns from an existing table.
 
-## Selection Actions
-Depending on a particular selection, the inspector panel will display actions that can be performed on the selection, as described above with columns. These selection actions will be determined by the type and number of items chosen.
+## Design Goals
 
-# User Flow
+- Allow users to split a table into multiple new tables by selecting columns
+- Help users understand the impact of splitting a table
+- Help users understand how the tables are linked after the split operation
 
-1. **The user imports a table from a file**
+## Splitting a Table by Extracting Columns
 
-    This user flow assumes that the starting point is an imported table containing columns that can be extracted and turned into entities. A column that may be transformed into a table or entity will usually include other columns that are attributes for that entity. If the database is not normalized, the entity to be extracted might include duplicated values. For example, consider a table of publications with multiple entries for the same publisher in the publisher column.
+### Selecting columns as a starting point
 
-2. **The user selects the columns (attributes) for the new entity**
+The user will select one or multiple columns to start with the column extraction process. All columns can be extracted, except for primary key columns. Once selected, the inspector panel will list 'Extract Columns to Table' as an action. Selecting this option will open the 'Extract Columns to Table' dialog.
```

Comment:
I would recommend calling this "Move Columns to Another Table" instead of "Extract Columns to Table", it's more clear.

Also, I assume we want the singular "Column" ("Move Column to Another Table") if there's only one column selected, I think it's worth specifying this.

Reply:
@kgodey My concern with moving vs. extraction is that moving suggests just changing the columns location, but we also create new columns and change the total rows of the moved data (deduplicate).

Reply:
I get your point but there's still something about the current phrasing that bugs me. Part of it is that you can either create a new table or move them to an existing table, so you're not always extracting the columns to a table. 

Maybe we can instead have two separate actions?
- Create new linked table using columns
- Move column to existing linked table

Reply:
I've updated the actions